pkg/catalogv2/system/system_test.go

```diff
@@ -490,7 +491,7 @@ func TestInstall(t *testing.T) {
 					Chart: &chart.Chart{Metadata: &chart.Metadata{Version: "100.0.0+up1.0.0"}},
 				}},
 			},
-			expected: repo.ErrNoChartName,
+			expected: errors.New("no chart version found for test-chart-102.0.0+up1.0.0"),
```

Comment:
Setting the [version](https://github.com/rancher/rancher/pull/42304/files#diff-96f7551d1bcafc092b3502857d33f73f576c753e3a4311b8660fabc446b383b0R275) to fetch via the indexer, instead of the latest version, when `minVersion` is set, means that `index.Get` now fails via a different code path in that case: instead of not finding any chart, it does not find that specific version of the chart. Hence the new error expectation.

Reply:
I don't think we should remove the comment since the function will not always return the version you requested, depending on the input. For instance, if you request rc1 and it does not exist, it will return rc2. 
We can update the comment to say that the function does not always return an error if the exact version is not found.